src/gamecontroller.cs

```diff
@@ -2170,5 +2202,499 @@ interface GCProductCategory
 
 		[Field ("GCProductCategoryKeyboard")]
 		NSString Keyboard { get; }
+
+		[iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+		[Field ("GCProductCategoryHID")]
+		NSString GCProductCategoryHid { get; }
+	}
+
+	[iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface GCRacingWheel : GCDevice {
+		[Static]
+		[Export ("connectedRacingWheels")]
+		NSSet<GCRacingWheel> ConnectedRacingWheels { get; }
+
+		[Export ("acquireDeviceWithError:")]
+		bool AcquireDevice ([NullAllowed] out NSError error);
+
+		[Export ("relinquishDevice")]
+		void RelinquishDevice ();
+
+		[Export ("acquired")]
+		bool Acquired { [Bind ("isAcquired")] get; }
+
+		[Export ("wheelInput", ArgumentSemantic.Strong)]
+		GCRacingWheelInput WheelInput { get; }
+
+		[Export ("snapshot")]
+		bool Snapshot { [Bind ("isSnapshot")] get; }
+
+		[Export ("capture")]
+		GCRacingWheel Capture { get; }
+
+		[Notification, Field ("GCRacingWheelDidConnectNotification")]
+		NSString DidConnectNotification { get; }
+
+		[Notification, Field ("GCRacingWheelDidDisconnectNotification")]
+		NSString DidDisconnectNotification { get; }
+	}
+
+	[iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (GCRacingWheelInputState))]
+	interface GCRacingWheelInput : GCDevicePhysicalInput {
+		// Sealed since GCDevicePhysicalInput.Capture returns IGCDevicePhysicalInputState
+		[Sealed, Export ("capture")]
+		GCRacingWheelInputState WheelInputCapture { get; }
+
+		// Sealed since GCDevicePhysicalInput.NextInputState returns NSObject
+		[Sealed, NullAllowed, Export ("nextInputState")]
+		IGCDevicePhysicalInputStateDiff WheelInputNextInputState { get; }
+	}
+
+	[iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	interface GCRacingWheelInputState : GCDevicePhysicalInputState {
+		[Export ("wheel")]
+		GCSteeringWheelElement Wheel { get; }
+
+		[NullAllowed, Export ("acceleratorPedal")]
+		IGCButtonElement AcceleratorPedal { get; }
+
+		[NullAllowed, Export ("brakePedal")]
+		IGCButtonElement BrakePedal { get; }
+
+		[NullAllowed, Export ("clutchPedal")]
+		IGCButtonElement ClutchPedal { get; }
+
+		[NullAllowed, Export ("shifter")]
+		GCGearShifterElement Shifter { get; }
+	}
+
+	[iOS (16,0), Mac (13,0), NoWatch, TV (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface GCSteeringWheelElement : GCAxisElement {
+		[Export ("maximumDegreesOfRotation")]
+		float MaximumDegreesOfRotation { get; }
+	}
+
+	interface IGCAxisElementName {}
```

Comment:
Whoever implemented this at Apple found a very complicated solution to something that looks quite simple.

I had a look at the headers, and it seems that these protocols is an attempt to group different NSString values together - basically creating an NSString-based enum of sorts 🤯

In any case, what I think we should do:

1. Don't bind any protocols that are defined with `objc_non_runtime_protocol`:

```objective-c

__attribute__((objc_non_runtime_protocol)) NS_REFINED_FOR_SWIFT
@protocol GCPhysicalInputElementName
@end

__attribute__((objc_non_runtime_protocol)) NS_REFINED_FOR_SWIFT
@protocol GCButtonElementName <GCPhysicalInputElementName>
@end

__attribute__((objc_non_runtime_protocol)) NS_REFINED_FOR_SWIFT
@protocol GCAxisElementName <GCPhysicalInputElementName>
@end

__attribute__((objc_non_runtime_protocol)) NS_REFINED_FOR_SWIFT
@protocol GCSwitchElementName <GCPhysicalInputElementName>
@end

__attribute__((objc_non_runtime_protocol)) NS_REFINED_FOR_SWIFT
@protocol GCDirectionPadElementName <GCPhysicalInputElementName>
@end
```

2. For each of these protocols, create a strong enum, and add the fields declared with each protocol to the corresponding enum.

For example:

```objective-c
GAMECONTROLLER_EXTERN GCInputButtonName GCInputButtonA NS_REFINED_FOR_SWIFT API_AVAILABLE(macos(11.0), ios(14.0), tvos(14.0));
GAMECONTROLLER_EXTERN GCInputButtonName GCInputButtonB NS_REFINED_FOR_SWIFT API_AVAILABLE(macos(11.0), ios(14.0), tvos(14.0));
GAMECONTROLLER_EXTERN GCInputButtonName GCInputButtonX NS_REFINED_FOR_SWIFT API_AVAILABLE(macos(11.0), ios(14.0), tvos(14.0));
GAMECONTROLLER_EXTERN GCInputButtonName GCInputButtonY NS_REFINED_FOR_SWIFT API_AVAILABLE(macos(11.0), ios(14.0), tvos(14.0));
```

would become:

```cs
	enum GCInputButtonName {
		[Field ("GCInputButtonA")]
		NSString A { get; }
		[Field ("GCInputButtonB")]
		NSString B { get; }
		[Field ("GCInputButtonX")]
		NSString X { get; }
		[Field ("GCInputButtonY")]
		NSString Y { get; }
		...
	}
```

@dalexsoto / @chamons what do you think?

The simpler solution is of course to just not bind these protocols for now, and decide later.

Reply:
I have to agree with @rolfbjarne this is way cleaner! the only thing is that it seems to be already done here as Fields

https://github.com/xamarin/xamarin-macios/pull/15692/files#diff-356008148e576743798c6d7e9adb15c283fdfe15e03fcae2edbe590828ccf533R1100-R1126

So probably this is not even needed? Also creating a test case may be the only way to figure out if the API is right. Maybe we can just leave them out, those protocols for now.

Reply:
@dalexsoto  @rolfbjarne 
I have removed the protocols that have the 'objc_non_runtime_protocol' attribute!